webpack/assets/javascripts/react_app/components/notifications/notifications.test.js

```diff
@@ -1,70 +1,50 @@
-import { mount } from 'enzyme';
 import React from 'react';
-import { generateStore } from '../../redux';
-import API from '../../redux/API/API';
-import { doesDocumentHasFocus } from '../../common/document';
-
-import { componentMountData, serverResponse } from './notifications.fixtures';
-
+import IntegrationTestHelper from '../../common/IntegrationTestHelper';
+import notificationReducer from '../../redux/reducers/notifications';
+import { get } from '../../redux/API/APIRequest';
+import { componentMountData } from './notifications.fixtures';
 import Notifications from './';
+import { APIMiddleware } from '../../redux/API';
+import { NOTIFICATIONS } from '../../redux/consts';
+import { IntervalMiddleware } from '../../redux/middlewares';
+import { registeredIntervalException } from '../../redux/middlewares/IntervalMiddleware/IntervalHelpers';
 
-jest.mock('../../redux/API/API');
-jest.mock('../../common/document');
+jest.mock('../../redux/API/APIRequest');
 
-let failResponse = { response: { status: 200 } };
+const notificationProps = {
+  data: componentMountData,
+};
 
-function mockjqXHR() {
-  return {
-    then: callback => {
-      callback(JSON.parse(serverResponse));
-      return mockjqXHR();
-    },
-    catch: failCallback => {
-      failCallback(failResponse);
-      return mockjqXHR();
-    },
-  };
-}
+const configureIntegrationHelper = () => {
+  const reducers = { notifications: notificationReducer };
+  const middlewares = [APIMiddleware, IntervalMiddleware];
+  return new IntegrationTestHelper(reducers, middlewares);
+};
 
 describe('notifications', () => {
-  doesDocumentHasFocus.mockImplementation(() => true);
-
   beforeEach(() => {
     global.tfm = {
       tools: {
         activateTooltips: () => {},
       },
     };
-    API.get = mockjqXHR;
-  });
-
-  it('full flow', () => {
-    const wrapper = mount(
-      <Notifications data={componentMountData} store={generateStore()} />
-    );
-    wrapper.find('.fa-bell').simulate('click');
-    expect(wrapper.find('.panel-group')).toHaveLength(1);
-    wrapper.find('.panel-group .panel-heading').simulate('click');
-    expect(wrapper.find('.unread')).toHaveLength(1);
-    wrapper.find('.unread').simulate('click');
-    expect(wrapper.find('.unread')).toHaveLength(0);
   });
```

Comment:
don't we need a full flow test?

Reply:
I will add it with the other tests refactor in #7138